configure.ac

```diff
@@ -1611,6 +1598,22 @@ else
 fi
 AC_SUBST([OS_ENVIRONMENT_PATH])
 
+dnl systemd-socket activation
+
+AC_ARG_WITH([systemd-socket], [AS_HELP_STRING([--with-systemd-socket[[=PATH]]], [support systemd socket activation])], [], [with_systemd_socket=check])
+
+dnl ########################################################################
+dnl systemd socket feature, only available if systemd init scripts requested
+dnl ########################################################################
+if test "x$with_systemd_service" != xno && test "x$with_systemd_socket" != xno
```

Comment:
moving this whole block AFTER with_systemd_service processing code allows us to include testing with_systemd_service and not check for systemd_socket if systemd_service is not requested.